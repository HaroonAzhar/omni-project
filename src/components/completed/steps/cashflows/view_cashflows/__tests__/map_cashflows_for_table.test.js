import mapCashflowsForTable from "../map_cashflows_for_table";
import e2eStatementResults from "../../utils/__tests__/data/statement_results";
import e2eAdjustments from "../../utils/__tests__/data/adjustments";

describe("mapCashflowsForTable", () => {
  it("Creates multiple entries for multiple adjustments", () => {
    const date = "2021-02-15";
    const statementResults = [
      {
        balance_adjustments: [100, -200],
        from: date,
        start_balance: 1000,
        interest: 50,
        to: "2021-02-16",
        end_balance: 950,
      },
    ];

    const adjustments = [
      {
        ActualDate: date,
        signedAmount: 100,
        Description: "bar",
      },
      {
        ActualDate: date,
        signedAmount: -200,
        Description: "foo",
        TransactionType: "transaction",
      },
    ];

    const expectedRows = [
      {
        Type: "Adjustment",
        Date: date,
        Amount: 100,
        Description: "bar",
        Balance: 1000,
      },
      {
        Type: "Adjustment",
        Date: date,
        Amount: -200,
        Description: "transaction - foo",
        Balance: 1000,
      },
      {
        Type: "Interest",
        Date: date,
        Description: " Interest 15/02/2021 - 16/02/2021",
        Balance: 950,
        Amount: 50,
      },
    ];

    const rows = mapCashflowsForTable({ statementResults, adjustments });

    expect(rows).toEqual(expectedRows);
  });

  it("Defaults to initial net loan amount, then exit fee", () => {
    const date = "2021-02-15";
    const end = "2021-02-16";

    const statementResults = [
      {
        balance_adjustments: [100],
        from: date,
        start_balance: 1000,
        interest: 50,
        to: end,
        end_balance: 1150,
      },
      {
        balance_adjustments: [100],
        from: end,
        start_balance: 1150,
        interest: 0,
        to: end,
        end_balance: 1250,
      },
    ];

    const adjustments = [];

    const expectedRows = [
      {
        Type: "Adjustment",
        Date: date,
        Amount: 100,
        Description: "Initial loan amount",
        Balance: 1000,
      },
      {
        Type: "Interest",
        Date: date,
        Description: " Interest 15/02/2021 - 16/02/2021",
        Balance: 1150,
        Amount: 50,
      },
      {
        Type: "Adjustment",
        Date: end,
        Amount: 100,
        Description: "Exit Fee",
        Balance: 1150,
      },
    ];

    const rows = mapCashflowsForTable({ statementResults, adjustments });

    expect(rows).toEqual(expectedRows);
  });

  it("e2e", () => {
    const actualRows = mapCashflowsForTable({
      statementResults: e2eStatementResults,
      adjustments: e2eAdjustments,
    });

    expect(actualRows).toMatchSnapshot();
  });
});
