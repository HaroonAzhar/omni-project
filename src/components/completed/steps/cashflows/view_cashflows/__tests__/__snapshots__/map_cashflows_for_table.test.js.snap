// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapCashflowsForTable e2e 1`] = `
Array [
  Object {
    "Amount": 303800,
    "Balance": 303928.3333333333,
    "Date": "2019-12-31T00:00:00",
    "Description": "Initial loan amount",
    "Type": "Adjustment",
  },
  Object {
    "Amount": 3850,
    "Balance": 307650,
    "Date": "2019-12-31T00:00:00",
    "Description": " Interest 31/12/2019 - 30/01/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 3850,
    "Balance": 311500,
    "Date": "2020-01-31T00:00:00",
    "Description": " Interest 31/01/2020 - 28/02/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 3850,
    "Balance": 315350,
    "Date": "2020-02-29T00:00:00",
    "Description": " Interest 29/02/2020 - 30/03/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 3850,
    "Balance": 319200,
    "Date": "2020-03-31T00:00:00",
    "Description": " Interest 31/03/2020 - 29/04/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 3850,
    "Balance": 323050,
    "Date": "2020-04-30T00:00:00",
    "Description": " Interest 30/04/2020 - 30/05/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 3850,
    "Balance": 326900,
    "Date": "2020-05-31T00:00:00",
    "Description": " Interest 31/05/2020 - 29/06/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 3850,
    "Balance": 330750,
    "Date": "2020-06-30T00:00:00",
    "Description": " Interest 30/06/2020 - 30/07/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 3850,
    "Balance": 334600,
    "Date": "2020-07-31T00:00:00",
    "Description": " Interest 31/07/2020 - 30/08/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 3850,
    "Balance": 338450,
    "Date": "2020-08-31T00:00:00",
    "Description": " Interest 31/08/2020 - 29/09/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 3850,
    "Balance": 342300,
    "Date": "2020-09-30T00:00:00",
    "Description": " Interest 30/09/2020 - 30/10/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 3850,
    "Balance": 346150,
    "Date": "2020-10-31T00:00:00",
    "Description": " Interest 31/10/2020 - 29/11/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 3593.3333333333335,
    "Balance": 349743.3333333333,
    "Date": "2020-11-30T00:00:00",
    "Description": " Interest 30/11/2020 - 27/12/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 478.90614896459584,
    "Balance": 350222.2394822979,
    "Date": "2020-12-28T00:00:00",
    "Description": " Interest 28/12/2020 - 29/12/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 350.57281563126253,
    "Balance": 350572.8122979292,
    "Date": "2020-12-31T00:00:00",
    "Description": "Extension Interest 31/12/2020 - 31/12/2020",
    "Type": "Interest",
  },
  Object {
    "Amount": 350.57281563126253,
    "Balance": 350923.3851135605,
    "Date": "2021-01-01T00:00:00",
    "Description": "Extension Interest 01/01/2021 - 01/01/2021",
    "Type": "Interest",
  },
  Object {
    "Amount": 350.57281563126253,
    "Balance": 351273.9579291917,
    "Date": "2021-01-02T00:00:00",
    "Description": "Extension Interest 02/01/2021 - 02/01/2021",
    "Type": "Interest",
  },
  Object {
    "Amount": 9114.893206412826,
    "Balance": 360388.85113560455,
    "Date": "2021-01-03T00:00:00",
    "Description": "Extension Interest 03/01/2021 - 28/01/2021",
    "Type": "Interest",
  },
  Object {
    "Amount": 350.57281563126253,
    "Balance": 360739.42395123583,
    "Date": "2021-01-29T00:00:00",
    "Description": "Extension Interest 29/01/2021 - 29/01/2021",
    "Type": "Interest",
  },
  Object {
    "Amount": 721.4788479024716,
    "Balance": 361460.9027991383,
    "Date": "2021-01-31T00:00:00",
    "Description": "Default Interest 31/01/2021 - 01/02/2021",
    "Type": "Interest",
  },
  Object {
    "Amount": -1000,
    "Balance": 361821.6422230895,
    "Date": "2021-02-02T00:00:00",
    "Description": "Interest Rebate (Manual)",
    "Type": "Adjustment",
  },
  Object {
    "Amount": 1000,
    "Balance": 361821.6422230895,
    "Date": "2021-02-02T00:00:00",
    "Description": "Further Advance on a Single Advance Loan",
    "Type": "Adjustment",
  },
  Object {
    "Amount": 360.7394239512358,
    "Balance": 361821.6422230895,
    "Date": "2021-02-02T00:00:00",
    "Description": "Default Interest 02/02/2021 - 02/02/2021",
    "Type": "Interest",
  },
  Object {
    "Amount": -1000,
    "Balance": 361936.13564704073,
    "Date": "2021-02-03T00:00:00",
    "Description": "Ad-Hoc Balance Reduction",
    "Type": "Adjustment",
  },
  Object {
    "Amount": -123,
    "Balance": 361936.13564704073,
    "Date": "2021-02-03T00:00:00",
    "Description": "Import Adjustment",
    "Type": "Adjustment",
  },
  Object {
    "Amount": -123,
    "Balance": 361936.13564704073,
    "Date": "2021-02-03T00:00:00",
    "Description": "Ad-Hoc Balance Reduction",
    "Type": "Adjustment",
  },
  Object {
    "Amount": -1000,
    "Balance": 361936.13564704073,
    "Date": "2021-02-03T00:00:00",
    "Description": "Import Adjustment",
    "Type": "Adjustment",
  },
  Object {
    "Amount": 1000,
    "Balance": 361936.13564704073,
    "Date": "2021-02-03T00:00:00",
    "Description": "Import Adjustment",
    "Type": "Adjustment",
  },
  Object {
    "Amount": 1000,
    "Balance": 361936.13564704073,
    "Date": "2021-02-03T00:00:00",
    "Description": "Litigation",
    "Type": "Adjustment",
  },
  Object {
    "Amount": 360.4934239512358,
    "Balance": 361936.13564704073,
    "Date": "2021-02-03T00:00:00",
    "Description": "Default Interest 03/02/2021 - 03/02/2021",
    "Type": "Interest",
  },
  Object {
    "Amount": -1000,
    "Balance": 361295.629070992,
    "Date": "2021-02-04T00:00:00",
    "Description": "Import Adjustment",
    "Type": "Adjustment",
  },
  Object {
    "Amount": 2156.9605437074147,
    "Balance": 363093.09619074815,
    "Date": "2021-02-04T00:00:00",
    "Description": "Default Interest 04/02/2021 - 09/02/2021",
    "Type": "Interest",
  },
  Object {
    "Amount": -1000,
    "Balance": 362451.5896146994,
    "Date": "2021-02-10T00:00:00",
    "Description": "Partial Redemption - Foo",
    "Type": "Adjustment",
  },
  Object {
    "Amount": -1000,
    "Balance": 362451.5896146994,
    "Date": "2021-02-10T00:00:00",
    "Description": "Redemption",
    "Type": "Adjustment",
  },
  Object {
    "Amount": 1000,
    "Balance": 362451.5896146994,
    "Date": "2021-02-10T00:00:00",
    "Description": "Ad-Hoc Cost",
    "Type": "Adjustment",
  },
  Object {
    "Amount": 358.4934239512358,
    "Balance": 362451.5896146994,
    "Date": "2021-02-10T00:00:00",
    "Description": "Default Interest 10/02/2021 - 10/02/2021",
    "Type": "Interest",
  },
  Object {
    "Amount": 123,
    "Balance": 361932.20603865065,
    "Date": "2021-02-11T00:00:00",
    "Description": "Import Adjustment",
    "Type": "Adjustment",
  },
  Object {
    "Amount": -1000,
    "Balance": 361932.20603865065,
    "Date": "2021-02-11T00:00:00",
    "Description": "Partial Redemption",
    "Type": "Adjustment",
  },
  Object {
    "Amount": 357.6164239512358,
    "Balance": 361932.20603865065,
    "Date": "2021-02-11T00:00:00",
    "Description": "Default Interest 11/02/2021 - 11/02/2021",
    "Type": "Interest",
  },
  Object {
    "Amount": 1000,
    "Balance": 363290.8224626019,
    "Date": "2021-02-12T00:00:00",
    "Description": "Ad-Hoc Cost",
    "Type": "Adjustment",
  },
  Object {
    "Amount": 1075.8492718537075,
    "Balance": 364008.05531050434,
    "Date": "2021-02-12T00:00:00",
    "Description": "Default Interest 12/02/2021 - 14/02/2021",
    "Type": "Interest",
  },
]
`;
